src/network/networkd-address.c

```diff
@@ -98,6 +98,9 @@ void address_free(Address *address) {
         if (address->link) {
                 set_remove(address->link->addresses, address);
                 set_remove(address->link->addresses_foreign, address);
+
+                if (in_addr_is_link_local(AF_INET6, &address->in_addr))
+                        memzero(&address->link->ipv6ll_address, sizeof(struct in6_addr));
```

Comment:
Yep, I can confirm that this fixes the issue in #3264.
But I don't think that `ipv6ll_address` should be nuked after any link local address is lost. For example I can do the following:
```
$ sudo ip a a dev wlp3s0 fe80::226:c6ff:fe85:a3c3/64
$ sudo ip a d dev wlp3s0 fe80::226:c6ff:fe85:a3c3/64
```
With you patch, `ipv6ll_address` will be zeroed. I think the check should use `in_addr_equal` to compare the address being removed with `ipv6ll_address` directly.